test/CMakeLists.txt

```diff
@@ -64,6 +64,11 @@ if(DOWNLOAD_GTEST OR GTEST_SRC_DIR)
 
     # Add googletest directly to our build. This defines
     # the gtest and gtest_main targets.
+    
+    if(MSVC)
+        set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
```

Comment:
I wasn't able to compile the unit test. The reason is the Visual Studio runtime library. This forces gtest to use the shared runtime library from Visual Studio.